zerver/actions/streams.py

```diff
@@ -507,6 +508,14 @@ def send_stream_creation_events_for_previously_inaccessible_streams(
     streams = [stream_dict[stream_id] for stream_id in stream_ids]
     setting_groups_dict: dict[int, int | AnonymousSettingGroupDict] | None = None
 
+    private_streams = [stream for stream in streams if not stream.is_public()]
+    if private_streams:
+        users_dict_with_metadata_access_via_permission_groups = (
+            get_users_dict_with_metadata_access_to_streams_via_permission_groups(
+                private_streams, realm.id
+            )
+        )
```

Comment:
So I think this won't work for when we add `can_join_group` and allow guests to join, since public channels can become accessible via permission groups too. 

Not a blocker for merging this, since we haven't added that setting yet and we plan to first add it with `allow_everyone_group=False`, but it may require some attention.